types/enigma.js/index.d.ts

```diff
@@ -0,0 +1,196 @@
+// Type definitions for enigma.js 2.2
+// Project: https://github.com/qlik-oss/enigma.js/
+// Definitions by: Konrad Mattheis <https://github.com/konne>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare module "enigma.js" {
```

Comment:
You shouldn't need to use the `declare module` syntax here, you can just unwrap all of this and write something like

```ts
const enigmaJS: IEnigmaClass;
export = enigmaJS;
export as namespace enigmaJS;
```

Reply:
@DanielRosenwasser thx, I already tried that, but if I do that I can not use the definied interfaces in enigmaJS namespace in my normal sourcecode that have imported that type file.

Reply:
For `export = e` syntax you need to import it with `import enigma = require('enigma.js')`. If you do this, does @DanielRosenwasser's suggestion work? (Note that his name suggestion of `enigmaJS: IEnigmaClass` will merge with the namespace `declare namespace enigmaJS`, so I would keep the name `e: IEnigmaJSClass`.)